main.py

```diff
@@ -197,8 +253,18 @@ def main(argv: Optional[Sequence[str]] = None):
     request_id = validate_response.json()["request_id"]
 
     submit_response = submit(token, request_id)
-    download_html_report(token, request_id, payload)
-    parse_results(submit_response.json())
+    download_and_save_html_report(token, request_id, payload)
+
+    # if this is true it compares the exceptions and results
+    if parse_results(submit_response.json()):
+        response_in_json = download_report(token, request_id, payload, "json")
+        response_in_json = read_bytes_response_as_dict(response_in_json)
+
+        failures = get_appinspect_failures_list(response_in_json)
+
+        expected_failures = parse_except_dict_to_list(read_yaml_as_dict(".appinspect.expect.yaml"))
```

Comment:
Let's have a different file for API failures.

Reply:
What do you mean different file? We want to have seperate files in our addon repos for CLI failures and API failures?

Reply:
We will use `.appinspect_api.expect.yaml` file as we agreed